src/ui/Base/RegisteredNamedMethods.ts

```diff
@@ -248,17 +248,26 @@ function getCoveoAnalytics(element: HTMLElement) {
  * service automatically converts white spaces to underscores, and uppercase characters to lowercase characters in key
  * names. Each value must be a simple string. If you do not need to log metadata, you can simply pass an empty JSON
  * ( `{}` ).
+ * @param result The IQueryResult that the custom event is linked to, if any.
  */
-export function logCustomEvent(element: HTMLElement, customEventCause: IAnalyticsActionCause, metadata: IStringMap<string>) {
+export function logCustomEvent(
+  element: HTMLElement,
+  customEventCause: IAnalyticsActionCause,
+  metadata: IStringMap<string>,
+  result: IQueryResult
+) {
   var client = getCoveoAnalyticsClient(element);
   if (client) {
-    client.logCustomEvent<any>(customEventCause, metadata, element);
+    client.logCustomEvent<any>(customEventCause, metadata, element, result);
   }
 }
 
-Initialization.registerNamedMethod('logCustomEvent', (element: HTMLElement, customEventCause: IAnalyticsActionCause, metadata: any) => {
-  logCustomEvent(element, customEventCause, metadata);
-});
+Initialization.registerNamedMethod(
+  'logCustomEvent',
+  (element: HTMLElement, customEventCause: IAnalyticsActionCause, metadata: any, result: IQueryResult) => {
```

Comment:
Similar comment here `result` should be optional so that it is backward compatible (aka: code calling this without a result before would not start erroring)

Reply:
Sure, my bad.